version-ranges/src/lib.rs

```diff
@@ -229,12 +229,16 @@ impl<V: Ord> Ranges<V> {
     }
 
     /// Returns true if self contains the specified value.
-    pub fn contains(&self, version: &V) -> bool {
+    pub fn contains<Q>(&self, version: &Q) -> bool
+    where
+        V: Borrow<Q>,
+        Q: ?Sized + PartialOrd,
+    {
         self.segments
             .binary_search_by(|segment| {
                 // We have to reverse because we need the segment wrt to the version, while
                 // within bounds tells us the version wrt to the segment.
-                within_bounds(version, segment).reverse()
+                within_bounds(version.borrow(), segment).reverse()
```

Comment:
Interesting, I expected you to need `segment.borrow()` here. Why does this work... hmmm.

Reply:
e.g., Below, you are doing `start.borrow()` and not `version.borrow()`, which is what I'd expect.

Reply:
Did you expect that I'd need _both_ `segment.borrw()` and `version.borrow()`?

Reply:
No, just `segment.borrow()`. I think I'd have to play around with this to figure it out.

Reply:
Oh ok. That's a hint that I can pull on then.

Reply:
Oooooo derp, yes. I was missing `within_bounds`, and _that's_ where you do the `borrow()` call and it is done on the thing I expect.

(This pattern from `HashMap` inverts the typical use of `Borrow`, where you call `borrow()` on the parameter to match what is stored. But here, you want to call `borrow()` on the thing stored to match what is provided.)